tests/unit/lms/services/canvas_test.py

```diff
@@ -209,20 +209,54 @@ def test_find_matching_file_in_course_returns_the_matching_file_id(
         ]
 
         matching_file_id = finder.find_matching_file_in_course(
-            sentinel.course_id, sentinel.file_id
+            sentinel.course_id, [sentinel.file_id]
         )
 
         file_service.get.assert_called_once_with(sentinel.file_id, type_="canvas_file")
         canvas_api_client.list_files.assert_called_once_with(sentinel.course_id)
         assert matching_file_id == str(sentinel.matching_file_id)
 
-    def test_find_matching_file_in_course_returns_None_if_theres_file_in_the_db(
+    def test_find_matching_file_in_course_with_multiple_file_ids(
+        self, finder, canvas_api_client, file_service
+    ):
+        matching_file = factories.File()
+        file_service.get.side_effect = [
+            # The first file_id isn't found in the DB.
+            None,
+            # The second file_id is in the DB but not found in the course.
+            factories.File(),
+            # The third file_id *will* be found in the course.
+            matching_file,
+        ]
+        canvas_api_client.list_files.return_value = [
+            {"id": 1, "display_name": "File 1", "size": 1024},
+            {
+                "id": sentinel.matching_file_id,
+                "display_name": matching_file.name,
+                "size": matching_file.size,
+            },
+        ]
+
+        matching_file_id = finder.find_matching_file_in_course(
+            sentinel.course_id,
+            [sentinel.file_id_1, sentinel.file_id_2, sentinel.file_id_3],
+        )
+
+        # It looked up each file_id in the DB in turn.
+        assert file_service.get.call_args_list == [
+            call(sentinel.file_id_1, type_="canvas_file"),
+            call(sentinel.file_id_2, type_="canvas_file"),
+            call(sentinel.file_id_3, type_="canvas_file"),
+        ]
+        assert matching_file_id == str(sentinel.matching_file_id)
```

Comment:
There's quite a lot going on in this test at once. I wonder if this is three different parameterised tests being run at once?

Reply:
I simplified it a tiny bit but it's not obvious to me how you'd make it simpler with parametrising